backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/logic/adapter/ItemStockSammMapper.java

```diff
@@ -0,0 +1,277 @@
+            if (!list.isEmpty()) {
+                material = list.get(0);
+                if (list.size() > 1) {
+                    log.warn("CustomerMaterialNumber " + matNbrCustomer + " is ambiguous, arbitrarily choosing " + material.getOwnMaterialNumber());
+                }
+            }
+        }
+        // Use MatNbrSupplier
+        if (material == null && matNbrSupplier != null) {
+            material = materialService.findByOwnMaterialNumber(matNbrSupplier);
+            if (matNbrCatenaX != null) {
+                log.warn("Unknown CatenaXNumber for Material " + material.getOwnMaterialNumber());
+            }
+            var mpr = mprService.find(material, partner);
+            if (mpr != null) {
+                if (!mpr.getPartnerMaterialNumber().equals(matNbrCustomer)) {
+                    log.warn("Unknown MaterialNumberCustomer " + matNbrCustomer + " for Material " + material.getOwnMaterialNumber());
+                }
+            }
+        }
+        if (material == null) {
+            log.warn("Could not identify material with CatenaXNbr " + matNbrCatenaX + " ,CustomerMaterialNbr " + matNbrCustomer + " and SupplierMaterialNbr " + matNbrSupplier);
+            return outputList;
+        }
+        for (var position : samm.getPositions()) {
+            Date lastUpdated = position.getLastUpdatedOnDateTime();
+            String supplierOrderId = null, customerOrderPositionId = null, customerOrderId = null;
+            if (position.getOrderPositionReference() != null) {
+                supplierOrderId = position.getOrderPositionReference().getSupplierOrderId();
+                customerOrderId = position.getOrderPositionReference().getCustomerOrderId();
+                customerOrderPositionId = position.getOrderPositionReference().getCustomerOrderPositionId();
+            }
+            for (var allocatedStock : position.getAllocatedStocks()) {
+                var builder = ReportedProductItemStock.builder();
+                var itemStock = builder
+                    .partner(partner)
+                    .material(material)
+                    .isBlocked(allocatedStock.getIsBlocked())
+                    .locationBpna(allocatedStock.getStockLocationBPNA())
+                    .locationBpns(allocatedStock.getStockLocationBPNS())
+                    .lastUpdatedOnDateTime(lastUpdated)
+                    .customerOrderId(customerOrderId)
+                    .supplierOrderId(supplierOrderId)
+                    .customerOrderPositionId(customerOrderPositionId)
+                    .measurementUnit(allocatedStock.getQuantityOnAllocatedStock().getUnit())
+                    .quantity(allocatedStock.getQuantityOnAllocatedStock().getValue())
+                    .build();
+                outputList.add(itemStock);
+            }
+        }
+        return outputList;
+    }
+
+    public List<ReportedMaterialItemStock> sammToReportedMaterialItemStock(ItemStockSAMM samm, Partner partner) {
```

Comment:
Did not evaluate this path but assume its similar as the ProductMaterialItemStock